src/components/Calculator.js

```diff
@@ -0,0 +1,186 @@
+import React from 'react';
+import './Calculator.css';
+import calculate from '../logic/calculator';
+
+class Calculator extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {};
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleClick = (buttonName) => {
+    this.setState((state) => calculate(state, buttonName));
+  };
+
+  displayResult = () => {
+    const { total, next, operation } = this.state;
+    const display = (total || '') + (operation || '') + (next || '');
+
+    return display || '0';
+  };
+
+  render() {
+    return (
+      <div className="calculator">
+        <p className="calculator-result">{this.displayResult()}</p>
+        <div className="calculator-buttons">
+          <button
+            name="AC"
+            type="button"
+            onClick={(e) => this.handleClick(e.target.name)}
+            className="grey"
+          >
+            AC
+          </button>
+          <button
+            name="+/-"
+            type="button"
+            onClick={(e) => this.handleClick(e.target.name)}
+            className="grey"
+          >
+            +/-
+          </button>
+          <button
+            name=""
+            type="button"
+            onClick={(e) => this.handleClick(e.target.name)}
+            className="grey"
+          >
+            %
```

Comment:
* Kindly fix the modulus logic as currently when an operation is performed let's say ( 70 % 10 ) it doesn't give the right answer